### ISSUE TITLE
Test to see if SymPhoTime can open generated PTU files

### ISSUE THREAD

**Issue by ekatrukha** [ekatrukha/PTU_Writer#1]

**Comment by ekatrukha** [comment]
Tried today with our SymPhoTime and it was not able to open it :(

And I couldn't find an option  to output any detailed error log. So no way to see what is the problem.

The first hypothesis is that some tags are missing.

In this case the easiest solution is to write all tags from an example file to the new PTU. And then remove them one by one and see which ones are crucial.

**Comment by ekatrukha** [comment]
Hello @NicoKiaru 
I've added some [example data](https://github.com/ekatrukha/PTU_Writer/tree/master/exampledata), one original from SymPhoTime, one converted. 
Maybe you can test and see if it can be opened by some python reader/opener

**Comment by ekatrukha** [comment]
Ok, I tried it myself, updated code and now this PTU can be loaded via [readPTU_FLIM](https://github.com/SumeetRohilla/readPTU_FLIM) python reader.

**Comment by NicoKiaru** [comment]
Hi @ekatrukha, 

So unfortunately the software I was interested in is the Leica software and I don't have the license anymore. Honestly I don't think I'll be able to have a look at it this year... but our goal is to get a new FLIM next year, so I'll definitely get back to the writer at this point - maybe sooner if I'm lucky.

Thanks again!

Cheers, 

Nico

**Comment by ekatrukha** [comment]
No problem, that's understandable.
We do not have Leica in-house, so I also cannot check if it reads the output. 
So far SymPhoTime fails without any meaningful message, so I am gonna close this issue.

Offtopic
Heads up, soon I plan to write you an email about BVV technical details, I was reading your code using it in `bdv-playground` and I have some questions.

**Comment by ekatrukha** [comment]
closed for now

**Comment by cgohlke** [comment]
The PTU tags are documented at [PicoQuant Unified Tag Dictionary](https://github.com/PicoQuant/PicoQuant-Time-Tagged-File-Format-Demos/blob/master/doc/apx_tagdictionary.html). The document specifies which tags are mandatory, for example, `Measurement_Mode` and `Measurement_SubMode`.

The [ptufile](https://github.com/cgohlke/ptufile/) library is also unable to read the `Example_image.sc_converted_v.0.0.2.ptu` file for the lack of those tags:

```
$ python -m ptufile Example_image.sc_converted_v.0.0.2.ptu
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "ptufile\__main__.py", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "ptufile\ptufile.py", line 2472, in main
    print(ptu)
  File "ptufile\ptufile.py", line 886, in __str__
    return indent(
           ^^^^^^^
  File "ptufile\ptufile.py", line 891, in <genexpr>
    if getattr(self, name) is not None
       ^^^^^^^^^^^^^^^^^^^
  File "ptufile\ptufile.py", line 1155, in measurement_mode
    return PtuMeasurementMode(self.tags['Measurement_Mode'])
                              ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
KeyError: 'Measurement_Mode'
```

**Comment by ekatrukha** [comment]
Hello Christoph @cgohlke,


Thank you for spending your time on testing the plugin!
Great, I did not know about mandatory tags. There should be `Submode` tag, I will add the rest and going to test again your library next week!

Cheers,
Eugene

**Comment by ekatrukha** [comment]
Hello Christoph @cgohlke ,

with the latest version bump (0.0.3) `ptufile` reads the generated file, thanks for your help!

The SymPhoTime still cannot open it. 
The problem seems to be in the `File_GUID` tag, I got an error "Cannot find file with this GUID".
Right now, I just copied its value from the original file.
But it seems I need to [generate](https://en.wikipedia.org/wiki/Universally_unique_identifier) it somehow.

Cheers,
Eugene
